examples/run_glue_tpu.py

```diff
@@ -256,14 +257,29 @@ def evaluate(args, model, tokenizer, prefix="", disable_logging=False):
             preds = np.squeeze(preds)
         result = compute_metrics(eval_task, preds, out_label_ids)
         results.update(result)
+        results['eval_loss'] = eval_loss.item()
+
+        # Average all metrics from each shard
+        def reduce_fn(results):
+            avg_result = defaultdict(int)
+            for result in results:
+                for k, v in result.items():
+                    avg_result[k] += v / float(len(results))
+            return avg_result
+        results = xm.mesh_reduce(
+            'eval_avg_metrics', results, reduce_fn)
 
         output_eval_file = os.path.join(eval_output_dir, prefix, "eval_results.txt")
-        with open(output_eval_file, "w") as writer:
-            logger.info("***** Eval results {} *****".format(prefix))
-            for key in sorted(result.keys()):
-                logger.info("  %s = %s", key, str(result[key]))
-                writer.write("%s = %s\n" % (key, str(result[key])))
-                tb_writer.add_scalar(key, result[key])
+        if xm.is_master_ordinal():
```

Comment:
is everything being logged here already on cpu?

Reply:
Yes.

Reply:
maybe let's add a comment? It's a subtle point that can be missed by code readers.